py3-jupyter-server.yaml

```diff
@@ -36,34 +16,108 @@ environment:
       - ca-certificates-bundle
       - nodejs
       - npm
-      - py3-setuptools
-      - python-3
+      - py3-supported-build-base
+      - py3-supported-hatch-jupyter-builder
+      - py3-supported-hatchling
       - wolfi-base
   environment:
     # This is needed to work around the error "ValueError: ZIP does not support timestamps before 1980"
     SOURCE_DATE_EPOCH: 315532800
 
+vars:
+  pypi-package: jupyter-server
+  import: jupyter_server
+
+data:
+  - name: py-versions
+    items:
+      3.10: '310'
+      3.11: '311'
+      3.12: '312'
+      3.13: '313'
+
 pipeline:
   - uses: git-checkout
     with:
       expected-commit: 535d7df3c0b358e31f514842c5509ca6ce64f056
       repository: https://github.com/jupyter-server/jupyter_server
       tag: v${{package.version}}
 
-  - name: Python Build
-    uses: python/build-wheel
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: ${{vars.pypi-package}} installed for python${{range.key}}
+    dependencies:
+      runtime:
+        - py${{range.key}}-anyio
+        - py${{range.key}}-argon2-cffi
+        - py${{range.key}}-jinja2
+        - py${{range.key}}-jupyter-client
+        - py${{range.key}}-jupyter-core
+        - py${{range.key}}-jupyter-events
+        - py${{range.key}}-jupyter-server-terminals
+        - py${{range.key}}-nbconvert
+        - py${{range.key}}-nbformat
+        - py${{range.key}}-overrides
+        - py${{range.key}}-packaging
+        - py${{range.key}}-prometheus-client
+        - py${{range.key}}-pyzmq
+        - py${{range.key}}-send2trash
+        - py${{range.key}}-terminado
+        - py${{range.key}}-tornado
+        - py${{range.key}}-traitlets
+        - py${{range.key}}-websocket-client
+      provides:
+        - py3-${{vars.pypi-package}}
```

Comment:
```suggestion

```
remove provides because it's covered in `-bin` compat and this approach is more backwards compatible because depends will get both binary as well as modules.